docs/settings.rst

```diff
@@ -58,6 +59,12 @@ ENCRYPT_BACKUP_CODES
 Backup codes to be encrypted before saving.
 Default: ``True``
 
+SECRET_KEY_LENGTH
+"""""""""""""""""
+Length of the shared secret key.  Please note: Google Authenticator requires a minimum length of 8 (16 on Android app) with a power of 2: 8, 16, 32...
```

Comment:
I'd like to verify this tomorrow. So if I set this to, say, 21, I should face some problems on iOS?

Reply:
@wgonczaronek yup!  You'll even face problems with the default of 20.  Please note the bug in iOS that @tallpol3 pointed out where it will accept things that it shouldn't if you use 'a' as the last character.

Reply:
Ahh good point!  If we are going to do a variable length we should probably cap it at say 96?  Can't image someone would want a secret that long though.  Maybe 48 would make more sense as an upper bounds?

Reply:
It's a library, so I'd rather avoid making this kind of assumptions regarding users' approach. On the other hand I can't think of anything more sensible than setting a huge limit and making a check on running the application whether the custom value fits limit (and throwing proper exception when necessary).

Reply:
So if the secret key has to be a power of 2 we have to go with 16 or 32.
8 is too short while anything above 32 is just overkill, thus I think there are 2 options. Either hardcode 16-length secret or give user option to choose 16/32. Both are easy to implement, second option gives more control to users though.

Reply:
Limiting it to either of those two options makes it a lot easier.  I'll update this to set the model to be 32 and then add a check in the validate method for RequestMFAMethodActivationSerializer that checks to make sure that setting is either 16 or 32 or raise a validation error.  With the docs stating to select either 16 or 32 and the /auth/app/activate/ endpoint throwing an exception that should be easy enough for people to figure out if they set things up incorrectly.  If there are any other gotchas please let me know and I can make adjustments as needed.  Cheers!